### ISSUE TITLE
undefined method `facets' for #<Spree::Search::Solr:0x007fb7af7a3f00> 

### ISSUE THREAD

**Issue by tavon** [romul/spree-solr-search#22]
This is Ash Ganatra's bug report posted in the mailing list:

After installing spree-solr-search I run the reindexing which runs fine but when I try to load the site I get this error

NoMethodError in Spree/home#index
Showing /Users/ashga/code/webportal/spree_products_filter/app/views/spree/shared/_taxonomies.html.erb where line #2 raised:

``` ruby
undefined method `facets' for #<Spree::Search::Solr:0x007fb7af7a3f00>
```

Extracted source (around line #2):

``` erb
1: <nav id="taxonomies" class="sidebar-item" data-hook><%
2: facets = @searcher.facets || []
3:   taxon_names = @taxon ? @taxon.self_and_descendants.map(&:name) : []
4:   for facet in facets  
5:     options = facet.options
Trace of template inclusion: /Users/ashga/.rbenv/versions/1.9.2-p290/lib/ruby/gems/1.9.1/bundler/gems/spree-68fee9e06a3f/core/app/views/spree/home/index.html.erb
```

**Comment by GeekOnCoffee** [comment]
tavon,

I'm not familiar enough with solr to be able to know for sure what's going on. The following change makes the error go away: 

app/views/spree/products/_facets.html.erb
@@ -1,5 +1,5 @@
 <%
-facets = @searcher.facets || []
+facets = @searcher.facets_hash || []

BUT I'm not sure if the functionality is actually still working once I make that change. Can you test it?

**Comment by jumph4x** [comment]
Above should take care of this.

**Comment by laurens** [comment]
It doesn’t.

In radar’s fix, two data structures are mixed up.

That `@searcher.facets_hash` seems to be the facets selected by the user. They are passed along the GET request.

In contrast `@searcher.facets` is the available facets returned by Solr with additional data like the name of the facets and the counters, which is used to build the facet list in the _facets.html.erb partial

This should be fixed by Pull Request #26.

**Comment by jumph4x** [comment]
Taken care of.
